tests/integration/config-aws-cli.sh

```diff
@@ -7,11 +7,29 @@ sudo snap install aws-cli --classic
 ACCESS_KEY=$(kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_ACCESS_KEY}' | base64 -d)
 SECRET_KEY=$(kubectl get secret -n minio-operator microk8s-user-1 -o jsonpath='{.data.CONSOLE_SECRET_KEY}' | base64 -d)
 
-S3_BUCKET="test"
+get_s3_endpoint(){
+    # Get S3 endpoint from MinIO
+    kubectl get service minio -n minio-operator -o jsonpath='{.spec.clusterIP}' 
+}
+
+# Wait for `minio` service to be ready and S3 endpoint to be available
+until get_s3_endpoint &> /dev/null ; do
+    echo "Waiting for minio service to be up and running..."
+    sleep 5
+done
+
+S3_ENDPOINT=$(get_s3_endpoint)
+
 DEFAULT_REGION="us-east-2"
 
 # Configure AWS CLI credentials
 aws configure set aws_access_key_id $ACCESS_KEY
 aws configure set aws_secret_access_key $SECRET_KEY
 aws configure set default.region $DEFAULT_REGION
+aws configure set endpoint_url "http://$S3_ENDPOINT"
+
+until aws s3 ls &> /dev/null; do
```

Comment:
I noticed that it takes a brief amount of time (something like 30 seconds or so) for the`aws cli` to work after setting up the credentials. I am not sure of what is causing that thing to happen at this moment (because this wasn't the case before). I spent a couple of days trying to understand this, and later opted for this workaround of waiting until, `aws s3 ls` command works.

Reply:
Added a definite retry logic.